### PR TITLE
Add new navigation and suggest a project page

### PR DESCRIPTION
**Pull request by jacobwindsor**
- Adds navigation bar
- Adds suggest a project page
- Creates new `pages` directories in modules to clarify what is a page and what is just a component. Hope you like this one!
- Adds a `visual-framework` module that contains some basic components that we can reuse across angular apps. The intention is to turn this into a library

### CODE REVIEW

**Review by MightyAx — APPROVED** [review]

File: src/app/services/heading.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
+import { HeadingService } from './heading.service';
+
+describe('HeadingServiceService', () => {
```
Comment (ke4) [inline src/app/services/heading.service.spec.ts:5]: I guess it is a typo

**Review by ke4 — APPROVED** [review]
I only find a small typo, but I was a bit rushed review, so it was only a quick walk through

**Review by ke4 — APPROVED** [review]

File: src/app/services/heading.service.spec.ts (hunk 1 above)
Comment (MightyAx) [reply src/app/services/heading.service.spec.ts]: looks fine to me, its a test, it runs fine

**Review by amnonkhen — APPROVED** [review]
Very nice!

**Review by amnonkhen — APPROVED** [review]

**Review by ke4 — APPROVED** [review]

**Review by ke4 — APPROVED** [review]

**Review by amnonkhen — APPROVED** [review]

**Pull request merged**